Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -128,69 +175,54 @@ void FilterEvents::init() {
   declareProperty("SpectrumWithoutDetector", "Skip", std::make_shared<StringListValidator>(spec_no_det),
                   "Approach to deal with spectrum without detectors. ");
 
-  declareProperty("SplitSampleLogs", true,
-                  "If selected, all sample logs will be splitted by the  "
-                  "event splitters.  It is not recommended for fast event "
-                  "log splitters. ");
+  declareProperty("DBSpectrum", EMPTY_INT(), "Spectrum (workspace index) for debug purpose. ");
 
-  declareProperty("NumberOutputWS", 0, "Number of output output workspace splitted. ", Direction::Output);
+  setPropertyGroup("FilterStartTime", titleAdditionalOptions);
+  setPropertyGroup("CorrectionToSample", titleAdditionalOptions);
+  setPropertyGroup("DetectorTOFCorrectionWorkspace", titleAdditionalOptions);
+  setPropertyGroup("IncidentEnergy", titleAdditionalOptions);
+  setPropertyGroup("SpectrumWithoutDetector", titleAdditionalOptions);
+  setPropertyGroup("DBSpectrum", titleAdditionalOptions);
 
-  declareProperty("DBSpectrum", EMPTY_INT(), "Spectrum (workspace index) for debug purpose. ");
+  //*********************
+  // Output Properties
+  //*********************
+  declareProperty("NumberOutputWS", 0, "Number of output output workspace splitted. ", Direction::Output);
 
   declareProperty(std::make_unique<ArrayProperty<string>>("OutputWorkspaceNames", Direction::Output),
                   "List of output workspaces names");
 
-  declareProperty("RelativeTime", false,
-                  "Flag to indicate that in the input Matrix splitting workspace,"
-                  "the time indicated by X-vector is relative to either run start time or "
-                  "some indicted time.");
-
-  declareProperty("FilterStartTime", "", "Start time for splitters that can be parsed to DateAndTime.");
-
-  declareProperty(std::make_unique<ArrayProperty<std::string>>("TimeSeriesPropertyLogs"),
-                  "List of name of sample logs of TimeSeriesProperty format. "
-                  "They will be either excluded from splitting if ExcludedSpecifiedLogs is "
-                  "specified as True. Or "
-                  "They will be the only TimeSeriesProperty sample logs that will be split "
-                  "to child workspaces.");
-
-  declareProperty("ExcludeSpecifiedLogs", true,
-                  "If true, all the TimeSeriesProperty logs listed will be "
-                  "excluded from duplicating. "
-                  "Otherwise, only those specified logs will be split.");
-
-  declareProperty("DescriptiveOutputNames", false,
-                  "If selected, the names of the output workspaces will "
-                  "include information about each slice.");
+  declareProperty("OutputUnfilteredEvents", false, "If selected, unfiltered events will be output.");
+  setPropertyGroup("OutputUnfilteredEvents", titleOutputWksp);
 }
 
 std::map<std::string, std::string> FilterEvents::validateInputs() {
```

Comment:
Should this method verify that the user only specified the filtering in a single way? It is more of a "nice to have" than a "must be done."

Reply:
Could you please clarify what you mean by a "single way"? Are you referring to the 3 supported types of splitter workspaces - Splitters, Table, Matrix?  In that case the answer would be: `OutputUnfilteredEvents` should work the same way regardless of how the splitter was specified.

Reply:
Sorry, I misunderstood your comment.  I wasn't looking at the last line of the code snippet!  Yes, that would be a nice check to have.  I also have the following question.  Currently `FilterEvents::validateInputs` checks things like: a table workspace must have 3 columns, a matrix workspace must be of histogram type, etc.  But now we have the same kind of checks in the corresponding `TimeSplitter` constructors.  Should we then delete these checks from `FilterEvents::validateInputs`?

Reply:
That's a bit more subtle. `validateInputs()` will put error messages in the gui before the algorithm starts doing its thing, but the constructors should definitely keep there checking. One way to leverage the constuctors is to use them in `validateInputs()` to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

Reply:
@peterfpeterson, I am back wondering what a "single way" means.  The user always provides just a single workspace describing splitters.  We then cast it to 3 different types of  workspace, and whichever pointer is valid - we then use it.

Reply:
> One way to leverage the constuctors is to use them in validateInputs() to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

Reply:
One way to leverage the constuctors is to use them in validateInputs() to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

Reply:
I think it is actually covered. It was a comment I put in before I had read through more of the code.

Reply:
One way to leverage the constuctors is to use them in validateInputs() to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

In this case I would rather not mix input validation with object initialization. I suggest to keep this "code duplication" as is.